sled-agent/src/bin/sled-agent.rs

```diff
@@ -98,7 +99,7 @@ async fn do_run() -> Result<(), CmdError> {
 
             // Configure and run the Bootstrap server.
             let bootstrap_config = BootstrapConfig {
-                id: config.id,
+                id: Uuid::new_v4(),
```

Comment:
I think this is probably beyond the scope of this PR but I wonder if the uuids used for Bootstrap and Sled Agent should come from some hardware identity.  Has that been discussed anywhere?

Reply:
Yeah, this is touched upon in https://github.com/oxidecomputer/omicron/issues/945, admittedly more in the context of the bootstrap address.